js/bot.js

```diff
@@ -102,13 +103,34 @@ function compareFollowersToFollowing () {
 }
 
 function processNotificationEvent(message) {
- const userId = message.data.account.id
+  const { status, account, type } = message.data
 
   // if a user follows the bot
-  if (message.data.type === 'follow') {
-    followUser(userId).then((result) => {
+  if (type === 'follow') {
+    followUser(account.id).then((result) => {
       console.info('Followed back: ', result)
-    }).catch(console.error)
+    })
+    .catch(console.error)
+  }
+
+  // if a user favourites the bot's toot
+  if (type === 'favourite') {
+
+    // check if it's a deletable toot
+    if (!status.content.includes(FAVOURITE_TOOT_TO_DELETE_STRING)) {
```

Comment:
This will stop the case where if we toot from the bot account. If they favourited us saying something it'd delete the toot lol

Reply:
😂 nice catch, hadn’t thought of that obvious slapstick failure mode

Reply:
Isn't checking that it's a direct post a better safeguard than text matching?